tests/v2/integration/catalogv2/ui_plugin_test.go

```diff
@@ -0,0 +1,369 @@
+	}}
+	res, _ := client.Get("https://localhost:8443/v1/uiplugins/uk-locale/0.1.0/plugin/uk-locale-0.1.0.umd.min.js")
+
+	w.Require().Equal(res.StatusCode, http.StatusOK)
+}
+
+// TestGetSingleUnauthorizedExtension Tests that the requests fails and returns 404 if the
+// extension requires authentication and the user is not authenticated
+func (w *UIPluginTest) TestGetSingleUnauthorizedExtension() {
+	defer w.resetSettings()
+	client := &http.Client{Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}}
+	res, _ := client.Get("https://localhost:8443/v1/uiplugins/clock/0.2.0/plugin/clock-0.2.0.umd.min.js")
+	w.Require().Equal(res.StatusCode, http.StatusNotFound)
+}
+
+func (w *UIPluginTest) WaitForConfigMap(namespace, name, latestVersion string) error {
+	return kwait.Poll(1*time.Second, 3*time.Minute, func() (done bool, err error) {
+		cfgMap, err := w.corev1.ConfigMaps(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+		w.Require().NoError(err)
+		gz, err := gzip.NewReader(bytes.NewBuffer(cfgMap.BinaryData["content"]))
+		w.Require().NoError(err)
+		defer gz.Close()
+		data, err := io.ReadAll(gz)
+		w.Require().NoError(err)
+		index := &repo.IndexFile{}
+		w.Require().NoError(json.Unmarshal(data, index))
+		index.SortEntries()
+		return index.Entries["rancher-aks-operator"][0].Version < latestVersion, nil
+	})
+}
+
+func (w *UIPluginTest) updateConfigMap(cfgMap *v1.ConfigMap) string {
+	gz, err := gzip.NewReader(bytes.NewBuffer(cfgMap.BinaryData["content"]))
+	w.Require().NoError(err)
+	defer gz.Close()
+	data, err := io.ReadAll(gz)
+	w.Require().NoError(err)
+	index := &repo.IndexFile{}
+	w.Require().NoError(json.Unmarshal(data, index))
+	index.SortEntries()
+	latestVersion := index.Entries["rancher-aks-operator"][0].Version
+	index.Entries["rancher-aks-operator"] = index.Entries["rancher-aks-operator"][1:]
+	marshal, err := json.Marshal(index)
+	w.Require().NoError(err)
+	var compressedData bytes.Buffer
+	writer := gzip.NewWriter(&compressedData)
+	_, err = writer.Write(marshal)
+	w.Require().NoError(err)
+	w.Require().NoError(writer.Close())
+	cfgMap.BinaryData["content"] = compressedData.Bytes()
+	return latestVersion
+}
+
+func (w *UIPluginTest) waitForChart(status rv1.Status, name string, previousVersion int) error {
+	t := 360
+	var app *rv1.App
+	err := kwait.Poll(time.Duration(500*time.Millisecond), time.Duration(t)*time.Second, func() (done bool, err error) {
+		app, err = w.catalogClient.Apps(namespace.UIPluginNamespace).Get(context.TODO(), name, metav1.GetOptions{})
+		e, ok := err.(*errors.StatusError)
+		if ok && errors.IsNotFound(e) {
+			return false, nil
+		}
+		if err != nil {
+			return false, err
+		}
+		if app.Spec.Info.Status == status && app.Spec.Version > previousVersion {
+			return true, nil
+		}
+		return false, nil
+	})
+	w.Require().NoError(err)
+	return err
+}
+
+func (w *UIPluginTest) updateManagementCluster() error {
```

Comment:
there seems to be lot of unused func..please remove.

Reply:
Dead code, removed